src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
+  }
+
+  public inefficientSort(arr: number[]): number[] {
+    return arr.sort((a, b) => {
+      if (a > b) return 1;
+      if (a < b) return -1;
+      return 0;
+    });
```

Comment:
PullProofAI

3. **Sorting Efficiency in `inefficientSort`:**
   - The `inefficientSort` method can be optimized by directly returning the result of the comparison function without explicit checks.
   - Simplify the sorting logic for better performance.

```diff
   public inefficientSort(arr: number[]): number[] {
29:-        return arr.sort((a, b) => {
30:-            if (a > b) return 1;
31:-            if (a < b) return -1;
32:-            return 0;
33:-        });
29:+        return arr.sort((a, b) => a - b);
   }
```



<!-- This is an auto-generated comment by PullProofAI -->